language/move-lang/src/parser/lexer.rs

```diff
@@ -0,0 +1,375 @@
+                let hex_len = get_hex_digits_len(&text[2..]);
+                if hex_len == 0 {
+                    // Fall back to treating this as a "0" token.
+                    (Tok::U64Value, 1)
+                } else {
+                    (Tok::AddressValue, 2 + hex_len)
+                }
+            } else {
+                (Tok::U64Value, get_decimal_digits_len(&text))
+            }
+        }
+        'A'..='Z' => {
+            let len = get_name_len(&text);
+            match &text[len..].chars().next() {
+                Some('<') => (Tok::MNameBeginArgsValue, len + 1),
+                _ => (Tok::MNameValue, len),
+            }
+        }
+        'a'..='z' | '_' => {
+            let len = get_name_len(&text);
+            match &text[len..].chars().next() {
+                Some('<') => (Tok::NameBeginArgsValue, len + 1),
+                _ => (get_name_token(&text[..len]), len),
+            }
+        }
+        '&' => {
+            if text.starts_with("&mut ") {
+                (Tok::AmpMut, 5)
+            } else if text.starts_with("&&") {
+                (Tok::AmpAmp, 2)
+            } else {
+                (Tok::Amp, 1)
+            }
+        }
+        '|' => {
+            if text.starts_with("||") {
+                (Tok::PipePipe, 2)
+            } else {
+                (Tok::Pipe, 1)
+            }
+        }
+        '=' => {
+            if text.starts_with("==") {
+                (Tok::EqualEqual, 2)
+            } else {
+                (Tok::Equal, 1)
+            }
+        }
+        '!' => {
+            if text.starts_with("!=") {
+                (Tok::ExclaimEqual, 2)
+            } else {
+                (Tok::Exclaim, 1)
+            }
+        }
+        '<' => {
+            if text.starts_with("<=") {
+                (Tok::LessEqual, 2)
+            } else {
+                (Tok::Less, 1)
+            }
+        }
+        '>' => {
+            if text.starts_with(">=") {
+                (Tok::GreaterEqual, 2)
+            } else {
+                (Tok::Greater, 1)
+            }
+        }
+        '%' => (Tok::Percent, 1),
+        '(' => (Tok::LParen, 1),
+        ')' => (Tok::RParen, 1),
+        '*' => (Tok::Star, 1),
+        '+' => (Tok::Plus, 1),
+        ',' => (Tok::Comma, 1),
+        '-' => (Tok::Minus, 1),
+        '.' => (Tok::Period, 1),
+        '/' => (Tok::Slash, 1),
+        ':' => (Tok::Colon, 1),
+        ';' => (Tok::Semicolon, 1),
+        '^' => (Tok::Caret, 1),
+        '{' => (Tok::LBrace, 1),
+        '}' => (Tok::RBrace, 1),
+        _ => {
+            return Err(ParseError::InvalidToken {
+                location: start_offset,
+            });
+        }
+    };
+
+    Ok((tok, len))
+}
+
+// Return the length of the substring matching [a-zA-Z0-9_]. Note that
+// this does not do any special check for whether the first character
+// starts with a number, so the caller is responsible for any additional
```

Comment:
OK, how do we make this [quirk](https://doc.rust-lang.org/reference/identifiers.html) more visible? perhaps make this a doc comment (`///`) so it prints in doc?

Reply:
The function is private within this file and rust-doc is only for public functions, right?

Reply:
huh, there seems to be an option for it, but it may not be worth buying into:
https://github.com/rust-lang/cargo/pull/5543